crates/vm/src/expr.rs

```diff
@@ -925,7 +923,15 @@ fn select_best_index<'a>(
 
     // Go through each operator and index,
     // consuming all field constraints that can be served by an index.
-    for (col_list, cmp) in [OpCmp::Eq, OpCmp::NotEq, OpCmp::Lt, OpCmp::LtEq, OpCmp::Gt, OpCmp::GtEq]
+    //
+    // NOTE: We do not consider `OpCmp::NotEq` at the moment.
+    // To provide such an index scan efficiently, we'd need to use two chained index scans
```

Comment:
I would remove this comment, since in general, you usually don't answer `!=` conditions using an index.